android/src/facebook/TiFacebookModule.java

```diff
@@ -345,13 +372,28 @@ public void logCustomEvent(String event, @Kroll.argument(optional = true) Double
 		}
 	}
 
+    @Kroll.method
+    public void logRegistrationCompleted(String registrationMethod)
+    {
+        Activity activity = TiApplication.getInstance().getCurrentActivity();
+        AppEventsLogger logger = AppEventsLogger.newLogger(activity);
+        Bundle paramBundle = new Bundle();
+        paramBundle.putString(AppEventsConstants.EVENT_PARAM_REGISTRATION_METHOD, registrationMethod);
+
+        if (logger != null) {
+            logger.logEvent(AppEventsConstants.EVENT_NAME_COMPLETED_REGISTRATION, paramBundle);
+        }
+    }
+
 	@Kroll.method
-	public void logPurchase(double amount, String currency)
+	public void logPurchase(double amount, String currency,
+                            @Kroll.argument(optional = true) KrollDict parameters)
 	{
 		Activity activity = TiApplication.getInstance().getCurrentActivity();
 		AppEventsLogger logger = AppEventsLogger.newLogger(activity);
+        Bundle paramBundle = parameters != null ? Utils.mapToBundle(parameters) : null;
```

Comment:
I don't think the null-check is required. `mapToBundle` already performs a null check.